torch/optim/adam.py

```diff
@@ -308,6 +308,8 @@ def step(self, closure=None):
         {_capturable_doc}
         {_differentiable_doc}
         {_fused_doc}
+    .. Note::
+        A prototype implementation of Adam and AdamW for MPS supports `torch.float32` and `torch.float16`.
```

Comment:
@janeyx99 do we usually document which backend supports fused?
Taking one more step, should we have, in the optimizer landing page in the doc, an overview table of which implementation we have and their stability level so that users can see at a glance what their options are?

Reply:
A centralized landing page is a good idea.

Reply:
Let's do it in a follow up, or do you prefer doing it in this PR?

Reply:
A followup is fine!

Reply:
I'm trying to make a PR for the table. I'd like to discuss the criteria that we regard an implementation as stable. If there is no existing standard, my initial proposal would be a 6 month period after the commit to regard it as stable. As a result, Adam and AdamW on CUDA would be considered as stable, while the rest are of beta. @janeyx99 What do you think?